### PR TITLE
[compiler][ez] Add types for Math.min, recursive global

### PR DESCRIPTION
**Pull request by mofeiZ**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #30643
* #30642
* __->__ #30641

### CODE REVIEW

**Review by josephsavona — APPROVED** [review]

File: compiler/packages/babel-plugin-react-compiler/src/HIR/Globals.ts [hunk 1]
```diff
@@ -455,11 +466,15 @@ for (const [name, type_] of TYPED_GLOBALS) {
...
 }
 
-// Recursive global type
+// Recursive global types
 DEFAULT_GLOBALS.set(
   'globalThis',
   addObject(DEFAULT_SHAPES, 'globalThis', TYPED_GLOBALS),
 );
+DEFAULT_GLOBALS.set(
+  'global',
```
Comment (kassens) [inline compiler/packages/babel-plugin-react-compiler/src/HIR/Globals.ts:475]: `window` too.

File: compiler/packages/babel-plugin-react-compiler/src/HIR/Globals.ts [hunk 2]
```diff
@@ -136,6 +136,17 @@ const TYPED_GLOBALS: Array<[string, BuiltInType]> = [
           returnValueKind: ValueKind.Primitive,
         }),
       ],
+      [
+        'min',
```
Comment (kassens) [inline compiler/packages/babel-plugin-react-compiler/src/HIR/Globals.ts:140]: whole bunch more `Math.*` stuff missing.

**Review by kassens — APPROVED** [review]

File: compiler/packages/babel-plugin-react-compiler/src/HIR/Globals.ts (hunk 1 above)
Comment (mofeiZ) [reply compiler/packages/babel-plugin-react-compiler/src/HIR/Globals.ts:475]: I debated on adding window but figured we should leave it out for now. I wonder if we want to handle `window` and `document` separately as people often use these for DOM-manipulation (global mutable state)

File: compiler/packages/babel-plugin-react-compiler/src/HIR/Globals.ts (hunk 2 above)
Comment (mofeiZ) [reply compiler/packages/babel-plugin-react-compiler/src/HIR/Globals.ts:140]: Makes sense, added min as `Math.min` and `Math.max` differences surprised me a few times. Will go ahead and add a few others

**Pull request merged**